src/test/java/com/google/sps/servlets/ContestServletTest.java

```diff
@@ -327,9 +331,97 @@ public void oneElection_contestMissingInDatastore_testDoGet() throws IOException
     ds.put(electionEntityOne);
 
     when(httpServletRequest.getParameter("electionId")).thenReturn("9999");
+    when(httpServletResponse.getWriter()).thenReturn(printWriter);
+
+    ContestsServlet contestServlet = new ContestsServlet();
+    contestServlet.doGet(httpServletRequest, httpServletResponse);
+
+    verify(printWriter)
+        .println(
+            "{\"contests\":[{\"name\":\"myFirstContest\",\"candidates\":[],\"description\":\"This contest is important.\","
+                + "\"source\":\"Voter Information Project\"}],\"referendums\":[]}");
+  }
+
+  @Test
+  public void oneElection_candidateMissingInDatastore_testDoGet() throws IOException {
+    Entity electionEntity = new Entity("Election");
+    Entity contestEntity = new Entity("Contest");
+
+    Entity firstCandidateEntity = new Entity("Candidate");
+    Entity secondCandidateEntity = new Entity("Candidate");
+
+    electionEntity.setPropertiesFrom(electionEntityOne);
+    contestEntity.setPropertiesFrom(contestEntityOne);
+
+    firstCandidateEntity.setPropertiesFrom(candidateEntityOne);
+    secondCandidateEntity.setPropertiesFrom(candidateEntityTwo);
+
+    DatastoreService ds = DatastoreServiceFactory.getDatastoreService();
+    long contestId = ds.put(contestEntity).getId();
+    long firstCandidateId = ds.put(firstCandidateEntity).getId();
+
+    // Delete the Entity from Datastore to mimic what happens at the beginning
+    // of a call to doGet() in ElectionServlet.
+    long secondCandidateId = ds.put(secondCandidateEntity).getId();
```

Comment:
I don't quite understand the point of this.  Aren't you just inserting something and then promptly deleting it?

Reply:
This is to imitate what would happen when we do a mass deletion of entities when calling ElectionServlet/InfoCardServlet. Additionally, I can't get a non-zero ID value for the entities (a zero value ID will throw an error when I try to add it to the Contests entity) unless I add to Datastore.